corehq/apps/userreports/expressions/specs.py

```diff
@@ -600,19 +600,21 @@ def __call__(self, item, evaluation_context=None):
     @staticmethod
     @ucr_context_cache(vary_on=('related_doc_type', 'doc_id',))
     def _get_document(related_doc_type, doc_id, evaluation_context):
+        domain = evaluation_context.root_doc['domain']
+        assert domain
         document_store = get_document_store_for_doc_type(
-            evaluation_context.root_doc['domain'], related_doc_type,
-            load_source="related_doc_expression")
+            domain, related_doc_type, load_source="related_doc_expression")
         try:
             doc = document_store.get_document(doc_id)
         except DocumentNotFoundError:
             return None
-        if evaluation_context.root_doc['domain'] != doc.get('domain'):
+        if domain != doc.get('domain'):
             return None
+        if related_doc_type == 'CommCareUser':
+            doc['user_data'] = CommCareUser.wrap(doc).get_user_data(domain).to_dict()
```

Comment:
> I guess user data just hasn't been part of UCRs since the migration?

That's right - this has been broken since the migration.

You can't do a related doc lookup on a web user, since they don't have a top level `domain` attribute.  For the same reason, they also don't have a `user_data` attribute (since it varies from one domain to the next), though I guess we could inject one in this context.

Reply:
For what it's worth, this means of directly accessing the `user_data` from the raw JSON has been broken to some degree for years - user data is only supposed to be accessed via helper methods, as fields controlled by the profile aren't stored directly in the user document.  That means that the profile wasn't accounted for in the API or in UCR.  I just broke it further :laughing: